src/vocallhout.h

```diff
@@ -49,6 +49,22 @@ CURRENT_STRUCT(VOResponse) {
   };
 };
 
+CURRENT_STRUCT(WSConfig) {
+  CURRENT_FIELD(host, std::string);
+  CURRENT_FIELD(port, uint16_t);
+  CURRENT_FIELD(n_threads, int);
+  CURRENT_FIELD(timeout_ms, int);
+  static WSConfig FromFields(std::string host = "0.0.0.0", uint16_t port = 8080,
```

Comment:
Nit: If you want it to be constructed this way, `CURRENT_DEFAULT_CONSTRUCTOR(WSConfig) = delete;` may come in handy.

Also thx for making this into a `CURRENT_STRUCT`. Could have been just a `struct` for the time being, so thank you!

Reply:
Actually, this alone won't work, since you still need to construct the instance, if only from the static function There are two ways:

1. Introduce some inner type, i.e. `struct ConstructSelf`, and add a constructor that takes it, `CURRENT_CONSTRUCTOR(WSConfig)(ConstructSelf) {}`. This way you could construct it as `WSConfig((WSConfig::ConstructSelf()))`.

2. Make the default constructor `private` or `protected` instead of `delete`-ing it.

Reply:
If i understood correctly the idea  - I just have added protected `CURRENT_DEFAULT_CONSTRUCTOR(...)`